app/components/header.tsx

```diff
@@ -1,11 +1,57 @@
-import { Text, Center } from '@chakra-ui/react';
+import { Text, Box, Heading, Avatar, Button, Flex, HStack } from '@chakra-ui/react';
+import { Link, Form } from '@remix-run/react';
+
+import { TriangleUpIcon, LockIcon } from '@chakra-ui/icons';
+import { useUser } from '~/utils';
 
 export default function Header() {
+  const user = useUser();
   return (
     <header>
-      <Center>
-        <Text fontSize="2xl">Hi! I am the header</Text>
-      </Center>
+      <Flex
+        backgroundColor="brand.500"
+        height="70px"
+        alignItems="center"
+        shadow="2xl"
+        border="2px"
+        fontSize="18"
+      >
+        <HStack width="50%" color="white" justifyContent="flex-start" gap="10" marginLeft="10">
+          <Link to={{ pathname: '/domains' }}>
+            <Flex alignItems="center">
+              <TriangleUpIcon marginRight="2" />
+              <Text>Custom Domain</Text>
+            </Flex>
+          </Link>
+
+          <Link to={{ pathname: '/certificate' }}>
+            <Flex alignItems="center">
+              <LockIcon marginRight="2" />
+              <Text>Certificate</Text>
+            </Flex>
+          </Link>
+        </HStack>
+        <Link to={{ pathname: '/' }}>
+          <Heading as="h1" size="2xl" color="white">
+            My.Custom.Domain
+          </Heading>
+        </Link>
+        <Flex
+          width="50%"
+          justifyContent="flex-end"
+          alignItems="center"
+          color="white"
+          gap="5"
+          marginRight="10"
+        >
+          <Text>{user.username}</Text>
+          <Form action="/logout" method="post">
```

Comment:
I recall telescope followed the same format or am I mistaken?

Reply:
![image](https://user-images.githubusercontent.com/32577022/218156775-b1999aa8-da51-41ce-8348-c4a2757d042d.png)

I replaced the previous iteration with a menu. Now the right side of the header should display the `username` and a `menu`. The menu will display the full name of the user capped at 17 characters, and a sign out button

Reply:
The user will know their name, I'm not sure we gain anything by showing it, esp. when it has to be cut-off like this.  I would just show the `Sign Out` menu item.

What is the purpose of showing the generic avatar icon a second time in the menu?

Reply:
The icon makes the menu more filled out, and the name is there to replaced the username, because I think it's better to have it outside the menu. I'll re-write it now to only have the sign out button

Reply:
I like the idea of the "filled-out" menu, but we have so little data to use, I think it ends up looking more empty by repeating.  Here's what GitHub does:

<img width="214" alt="Screenshot 2023-02-10 at 12 43 08 PM" src="https://user-images.githubusercontent.com/427398/218159954-3862b432-dc97-4662-adca-963f76ada2f8.png">

In contrast to Teams and Slack, which use the image to better effect:

<img width="338" alt="Screenshot 2023-02-10 at 12 42 48 PM" src="https://user-images.githubusercontent.com/427398/218159961-0d25d089-89c9-4a49-bb0d-5566705f0a64.png">
<img width="350" alt="Screenshot 2023-02-10 at 12 42 41 PM" src="https://user-images.githubusercontent.com/427398/218159963-10a434f8-6344-460b-8b2d-fc8516cd1e7b.png">

I don't know if we'll ever have more than "Sign Out" or not?  But for sure we'll never have an image for all users, so I wouldn't rely on a design that tries to use it.

Reply:
Maybe we can put the "ITS Acceptable Use Policy" link in there, or help links, not sure...

Reply:
Yeah you do have a point, I replaced it with a single menu item
![image](https://user-images.githubusercontent.com/32577022/218160931-919f4f10-a4aa-4aff-a8cc-06f4257e628a.png)

Reply:
We could consider moving "Domains" and "Certificate" into that menu as well, which is kind of what GitHub does.

Reply:
That would leave the header less populated, unless you're referring to adding another menu for only the two which is similar to the original figma design

Reply:
I mean adding them to the menu you're making:

```
Domains
Certificate
-----------
Sign out
```

We don't have to put it there, just thinking out loud.

Reply:
Hmm, I think that would make add an extra step to the navigation. I think if there are more pages to navigate to, it would be good. But we only have domains and certificate